server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -135,6 +163,64 @@ class LivyServer extends Logging {
     sys.props("livy.server.serverUrl") = _serverUrl.get
   }
 
+  def runKinit(): Boolean = {
+    val keytab = livyConf.get(LAUNCH_KERBEROS_KEYTAB)
+    val principal = SecurityUtil.getServerPrincipal(livyConf.get(LAUNCH_KERBEROS_PRINCIPAL),
+      server.host)
+    require(principal != null,
+      s"Kerberos requires ${LAUNCH_KERBEROS_PRINCIPAL.key} to be provided.")
+    require(keytab != null,
+      s"Kerberos requires ${LAUNCH_KERBEROS_KEYTAB.key} to be provided.")
+    val cacheFileName = kinitCacheDir + "/krb5cc_0"
+    val commands = Seq("kinit", "-kt", keytab, "-c", cacheFileName, principal)
+    val proc = new ProcessBuilder(commands: _*).inheritIO().start()
+    proc.waitFor() match {
+      case 0 =>
+        debug("Ran kinit command successfully.")
+        livyConf.set(RSCConf.Entry.KINIT_CACHE_FILE, cacheFileName)
```

Comment:
Do you need to set this every time `runKinit` is called? Seems like you could do this initialization once and be done with it.